src/CP/valueselection/learning/rewards/generalreward.jl

```diff
@@ -0,0 +1,122 @@
+"""
+function set_reward!(::Type{StepPhase}, lh::LearnedHeuristic{SR, GeneralReward, A}, model::CPModel, symbol::Union{Nothing, Symbol}) where {
+    SR <: AbstractStateRepresentation,
+    A <: ActionOutput
+}
+    nothing
+end
+
+"""
+    set_reward!(::DecisionPhase, lh::LearnedHeuristic{GeneralReward, O}, model::CPModel)
+
+Change the current reward at the DecisionPhase. This is called right before making the next decision, so you know you have the very last state before the new decision
+and every computation like fixPoints and backtracking has been done.
+"""
+function set_reward!(::Type{DecisionPhase}, lh::LearnedHeuristic{SR, GeneralReward, A}, model::CPModel) where {
+    SR <: AbstractStateRepresentation,
+    A <: ActionOutput
+}
+    if !lh.firstActionTaken
+        #println("First action being taken")
+        lh.reward.value = 0
+    else
+        lh.reward.value = lh.reward.gamma*(model.statistics.lastPruning/(lh.reward.initialNumberOfVariableValueLinks - length(branchable_variables(model))))^lh.reward.beta
+        #println(model.statistics.lastPruning)
+        #println(lh.reward.initialNumberOfVariableValueLinks - length(branchable_variables(model)))
+        #println(string(lh.reward.value))
+        if lh.reward.value>0.001
+            #println("Variable part: "*string(lh.reward.value))
+        end
+        if !isnothing(model.objective)
+            lh.reward.value += -(model.statistics.objectiveDownPruning/(lh.reward.initMax - lh.reward.initMin)) + (model.statistics.objectiveUpPruning/(lh.reward.initMax - lh.reward.initMin))
+            #println("Objective part: "*string(-(model.statistics.objectiveDownPruning/(lh.reward.initMax - lh.reward.initMin)) + (model.statistics.objectiveUpPruning/(lh.reward.initMax - lh.reward.initMin))))
+            if -(model.statistics.objectiveDownPruning/(lh.reward.initMax - lh.reward.initMin)) + (model.statistics.objectiveUpPruning/(lh.reward.initMax - lh.reward.initMin)) != 0
+                #println("Objective domain: "*string(model.objective.domain))
+                #println("Down pruning: "*string(model.statistics.objectiveDownPruning))
+                #println("Up pruning: "*string(model.statistics.objectiveUpPruning))
+            end
+        end
+        #println(lh.reward.value)
+    end
+end
+
+
+"""
+    set_reward!(::EndingPhase, lh::LearnedHeuristic{GeneralReward, A}, model::CPModel, symbol::Union{Nothing, Symbol})
+
+Increment the current reward at the EndingPhase. Called when the search is finished by an optimality proof or by a limit in term of nodes or 
+in terms of number of solution. This is useful to add some general results to the reward like the number of ndoes visited during the episode for instance. 
+"""
+function set_reward!(::Type{EndingPhase}, lh::LearnedHeuristic{SR, GeneralReward, A}, model::CPModel, symbol::Union{Nothing, Symbol}) where {
+    SR <: AbstractStateRepresentation, 
+    A <: ActionOutput
+}
+    nqueens_conflict_counter = false
```

Comment:
This field is now useless as far as I can tell because never updated and seems to me to be inherited from an experimental reward proper to nqueens.